src/realm/collection_parent.hpp

```diff
@@ -68,54 +68,148 @@ enum class UpdateStatus {
     NoChange,
 };
 
+struct PathElement {
+    union {
+        std::string string_val;
+        int64_t int_val;
+    };
+    enum Type { string, integer } m_type;
 
-class PathElement : private Mixed {
-public:
     PathElement(int ndx)
-        : Mixed(int64_t(ndx))
+        : int_val(ndx)
+        , m_type(Type::integer)
     {
         REALM_ASSERT(ndx >= 0);
     }
     PathElement(size_t ndx)
-        : Mixed(int64_t(ndx))
+        : int_val(int64_t(ndx))
+        , m_type(Type::integer)
+    {
+    }
+
+    PathElement(StringData str)
+        : string_val(str)
+        , m_type(Type::string)
     {
     }
-    PathElement(const char* key)
-        : Mixed(key)
+    PathElement(const char* str)
+        : string_val(str)
+        , m_type(Type::string)
     {
     }
-    PathElement(StringData key)
-        : Mixed(key)
+    PathElement(const PathElement& other)
+        : m_type(other.m_type)
     {
+        if (other.m_type == Type::string) {
+            new (&string_val) std::string(other.string_val);
+        }
+        else {
+            int_val = other.int_val;
+        }
     }
+
+    PathElement(PathElement&& other) noexcept
+        : m_type(other.m_type)
+    {
+        if (other.m_type == Type::string) {
+            new (&string_val) std::string(std::move(other.string_val));
+        }
+        else {
+            int_val = other.int_val;
+        }
+    }
+    ~PathElement()
+    {
+        if (m_type == Type::string) {
+            string_val.std::string::~string();
+        }
+    }
+
     bool is_ndx() const noexcept
     {
-        return is_type(type_Int);
+        return m_type == Type::integer;
     }
     bool is_key() const noexcept
     {
-        return is_type(type_String);
+        return m_type == Type::string;
     }
 
     const size_t* get_if_ndx() const noexcept
     {
-        return reinterpret_cast<const size_t*>(get_if<Int>());
+        if (is_ndx()) {
+            return reinterpret_cast<const size_t*>(&int_val);
+        }
+        return nullptr;
     }
-    const StringData* get_if_key() const noexcept
+    const std::string* get_if_key() const noexcept
```

Comment:
Same as above here.

Reply:
The idea was to have an interface similar to std::variant::get_if<T>(),

Reply:
OK, my preference would be to have a wrapper around this ptr, but either with `std::optional`  or a naked ptr that code using this method will look the same:
```
auto key = colleciton.get_if_key()
if(key) {
```

Reply:
I have removed the functions. Not really needed